### PR TITLE
[LA Web] Fix animations in production bundle

### PR DESCRIPTION
**Pull request by m-bert**
## Summary

As pointed in #5561, layout animations don't work with production bundle. It is caused by fragile method of obtaining animation name, i.e. using `config.name` if config is function. During bundling, name of this function may change (in linked issue to `c`, in my case to `r`), so instead of animation user will see warning that reanimated couldn't load animation. 

This PR adds static field `className` to animations builders in order to provide access to animation name, even if config name gets changed during bundling.  

## Test plan

1.  Create new app (`npx create-expo-app`)
2. Install Reanimated from package (`npm pack` -> `yarn add <path_to_package>`) 
3. `npx expo export -p web`
4. `npx serve dist --single`

I used reanimated from local package to make sure that `expo export` works correctly.

<details>
<summary> Tested on the following code </summary>

```jsx
import Animated, { FadeOut, FadeOutRight } from "react-native-reanimated";
import { View, Button } from "react-native";
import { useState } from "react";

export default function AnimatedStyleUpdateExample(props) {
  const [show, setShow] = useState(true);

  return (
    <View
      style={{
        flex: 1,
        alignItems: "center",
        justifyContent: "center",
        flexDirection: "column",
      }}
    >
      {show && (
        <Animated.View
          style={[{ width: 100, height: 80, backgroundColor: "black" }]}
          exiting={FadeOut}
        />
      )}

      <Button
        title="toggle"
        onPress={() => {
          setShow((prev) => !prev);
        }}
      />
    </View>
  );
}

```

</details>

### CODE REVIEW

File: src/reanimated2/layoutReanimation/web/animationsManager.ts [hunk 1]
```diff
@@ -89,9 +89,15 @@ function tryGetAnimationConfigWithTransform<
     return null;
   }
 
+  type ConstructorWithStaticContext = {
```
Comment (tjzel) [inline src/reanimated2/layoutReanimation/web/animationsManager.ts:92]: This should use imported `LayoutAnimationStaticContext`.

**Review by tjzel — CHANGES_REQUESTED** [review]
As discussed, let's change `className` to something more relevant like `presetName`.

File: src/reanimated2/layoutReanimation/web/animationsManager.ts (hunk 1 above)
Comment (m-bert) [reply src/reanimated2/layoutReanimation/web/animationsManager.ts]: Both done in [fd57b63](https://github.com/software-mansion/react-native-reanimated/pull/5562/commits/fd57b63bbddc9d9a55fe555a828f6040b3814072).

**Review by tjzel — APPROVED** [review]
🚢 🇮🇹

**Pull request merged**